`TypeError` in `depots.fetch_codes()`

## Description
The `test_fetch_codes` method fails with a `TypeError: 'NoneType' object is not iterable` when `update=True`. The error occurs during dictionary comprehension when processing `depot_data`.

## Error traceback
```
tests\test_depot.py:25 (TestDepots.test_fetch_codes[True])
self = <tests.test_depot.TestDepots object at ...>
depots = <pyrcs.other_assets.depot.Depots object at ...>
update = True

    @pytest.mark.parametrize('update', [True, False])
    def test_fetch_codes(self, depots, update):
>       depots_codes = depots.fetch_codes(update=update, verbose=True)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

tests\test_depot.py:28: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pyrcs\other_assets\depot.py:680: in fetch_codes
    self.KEY: {next(iter(x)): next(iter(x.values())) for x in depot_data},
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

.0 = <list_iterator object at ...>

>       self.KEY: {next(iter(x)): next(iter(x.values())) for x in depot_data},
                        ^^^^^^^
        self.KEY_TO_LAST_UPDATED_DATE:
            max(next(itertools.islice(iter(x.values()), 1, 2)) for x in depot_data)
    }
E   TypeError: 'NoneType' object is not iterable

pyrcs\other_assets\depot.py:680: TypeError
```

## Expected Behaviour
The `fetch_codes` method should handle cases where `depot_data` contains `None` values or is empty gracefully, without throwing a `TypeError`.

## How to reproduce
```bash
pytest tests/test_depot.py::TestDepots::test_fetch_codes[True] -v
```

--- a/pyrcs/other_assets/depot.py
+++ b/pyrcs/other_assets/depot.py
@@ -482,15 +482,18 @@ class Depots(_Base):
 
         span_tags = soup.find_all(name='span', attrs={'class': 'tab2'})
         num_codes_dict = dict([
-            (int(span_tag.text), str(span_tag.next_sibling).replace(' = ', '').strip())
+            (int(span_tag.text), str(span_tag.next_sibling).replace('=', '').strip())
             for span_tag in span_tags])
 
-        numerical_codes.rename(columns={'sort by division': 'Division'}, inplace=True)
-        numerical_codes.Division = numerical_codes.Code.map(
-            lambda x: num_codes_dict[int(str(x)[-1])])
+        temp = numerical_codes.iloc[:, 0].str.split(' ', expand=True)
+        temp.columns = ['Code', 'Division']
+        temp.loc[:, 'Division'] = temp['Division'].map(lambda x: num_codes_dict[int(str(x)[-1])])
+        numerical_codes = pd.concat(
+            [temp, numerical_codes.drop(columns=numerical_codes.columns[0])], axis=1)
 
         h3_titles = [h3.text for h3 in soup.find_all('h3')]
         gwr_depot_codes_data = dict(zip(h3_titles, [alphabetical_codes, numerical_codes]))
+        gwr_depot_codes_data.update({'Keys to numerical codes': num_codes_dict})
 
         gwr_depot_codes = {
             self.KEY_TO_GWR: gwr_depot_codes_data,

